src/services/ProductExports.ts

```
@@ -0,0 +1,134 @@
+  const result: api.ProdExport = {
+    uuid: dbProdExport.productexportuuid,
+    transportId: dbProdExport.transportid,
+    datetime: dbProdExport.datetime,
+    productType: dbProdExport.productname,
+    amountOfProduct: dbProdExport.amountofproduct,
+    lastModified: dbProdExport.lastmodified,
+  };
+
+  return result;
+}
+
+
+/**
+ * Connector for the API and database to get all product exports
+ */
+export async function getProdExportsFromDb(): Promise<api.ProdExport[]> {
+  const dbProdExports: db.ProdExportDb[] = await db.getProdExports();
+  const results: api.ProdExport[] = [];
+
+  logger.info('results', results);
+
+  dbProdExports.forEach(function (item: db.ProdExportDb) {
+    const newExport: api.ProdExport = {
+      uuid: item.productexportuuid,
+      transportId: item.transportid,
+      datetime: item.datetime,
+      productType: item.productname,
+      amountOfProduct: item.amountofproduct,
+      lastModified: item.lastmodified,
+    };
+
+    results.push(newExport);
+  });
+
+  return results;
+}
+
+/**
+ * Connector for the API and database to create a new product export
+ * Throws error on invalid request
+ * Throws error with message "unhandled error" for unhandled errors
+ */
+export async function createProdExportInDb(apiReq: api.ProdExportReq): Promise<string> {
+  let productTypeId: number = -1;
+  const invalidFields: string[] = [];
+
+  // validate request
+  try {
+    productTypeId = await productTransactionsDb.getProductId(apiReq.productType);
+  } catch (e) {
+    throw new Error('Product type ' + apiReq.productType + ' not supported');
+  }
+
+  if (!apiReq.uuid) {
+    invalidFields.push('uuid');
+  }
+  if (!apiReq.transportId) {
+    invalidFields.push('transportId');
+  }
+  if (!apiReq.datetime) {
+    invalidFields.push('datetime');
+  }
+  if (!apiReq.productType) {
+    invalidFields.push('productType');
+  }
+  if (!apiReq.amountOfProduct) {
+    invalidFields.push('amountOfProduct');
+  }
+
```

if we know all the invalid fields by this point, shouldnt we throw an error now instead of trying to insert a request?

---

The error thrwoing is being done before the actual insert request is done. But I created the insert request struct before throwing the error, I'll switch them around.